src/libmongoc/src/mongoc/mongoc-topology-scanner.c

```diff
@@ -461,6 +472,7 @@ mongoc_topology_scanner_add (mongoc_topology_scanner_t *ts,
    node->ts = ts;
    node->last_failed = -1;
    node->last_used = -1;
+   node->hello_ok = hello_ok;
```

Comment:
Is this situation possible?

1. Initially a scanner node is added with `hello_ok` set to true.
2. A network error during a monitoring check with `hello` during a downgrade.
3. The subsequent monitoring check still uses `hello`.

If so, we may need to:
1. Reset `hello_ok` whenever a connection is created (`_begin_hello_cmd` may be a good spot to do so)
2. Update `hello_ok` by parsing in `_async_success`.

Reply:
I'm not familiar enough with this. I'd assume that the network error during monitoring check triggers the check in L329 of this file, where we don't call the monitoring command but rather the handshake command if the node failed recently. In this case, we'll revert to using the legacy hello command unless an API Version was declared on the client.

Reply:
Ah good point! I think you are correct. And the mock server tests verify this.